LICENSE

```diff
@@ -1,6 +1,13 @@
-MIT License
+Dual License of MIT or LGPL
```

Comment:
This is wrong. What you describe sounds like AND not OR, since a consumer of this project would have to comply with both sets of license conditions. 

Also, if you intend to go this route you need to be much more explicit about which files are under which licenses. You will also need to argue that there is a clear boundary between the two works and that your new code in pynose is not a derivative of the LGPL code copied from nose.

It would be far, far, far easier to just accept that the whole project should be under the LGPL.

Reply:
It cannot be **all** LGPL, as some of the new code is MIT, and those MIT parts cannot be changed.

Reply:
In that case I recommend that you identify the license on a file by file basis. I'm a huge advocate of the https://reuse.software/ specification, that should give you some clear guidance on what to do here.

Reply:
the whole work is effectively under LGPL; this is the nature of the license.

Reply:
> It cannot be **all** LGPL, as some of the new code is MIT, and those MIT parts cannot be changed.

If those new parts "can be reasonably considered independent and separate works in themselves", then you can publish them under MIT "when you distribute them as separate works" (quoting from section 2 of [the LGPL-2](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html#SEC1)). But if you publish the new code as part of a single library (called, say, "pynose"), then it must be under the LPGL.

If that is not possible, then you can't legally distribute this software at all.

Reply:
File-by-file is tricky, as both old code and new code can be in one file.

Reply:
@mauke This is why I'm advocating for a file-by-file identification of license. If the new work is so intertwined that it can't be separated on that basis then it's hard to argue that there are two separate works.

Reply:
> It cannot be **all** LGPL, as some of the new code is MIT, and those MIT parts cannot be changed.

If you wrote that new code, then you can license it as LGPL at any time as the copyright holder. For code you didn’t write, the MIT licence gives you explicit textual permission to *sublicense* under another licence, such as LGPL, as long as you retain the copyright notice and MIT licence.

Reply:
It's an interesting situation. The original LGPL code and the newer MIT code are intertwined. The original LGPL code cannot be relabled MIT, yet the MIT code cannot be relabeled to LGPL either. And the code may continue to change. Since we're living in modern times, I could put the following in:

"To differentiate between LGPL and MIT code, use `git diff` between `nose` `1.3.7` and the version of `pynose` that you're using."

Reply:
@mdmintz Read https://en.wikipedia.org/wiki/Derivative_work

Though at this point I recommend getting advice from a lawyer that specialises in open source licensing.

Reply:
> yet the MIT code cannot be relabeled to LGPL either

As I said in my previous comment, this is not really true. But if you don’t want to include the MIT text at all, there is a simple resolution; according to the commit log, the only contributors to this fork are you and @cclauss. The changes by the latter seem like they might not be eligible for copyright, but to eliminate ambiguity, you can ask @cclauss to agree to relicense their changes under LGPL. If agreed, you can label this fork purely as LGPL with no MIT notice at all.

Reply:
For what it’s worth I would also strongly encourage you to restore the contents of the [`AUTHORS` file](https://github.com/nose-devs/nose/blob/7c26ad1e6b7d308cafa328ad34736d34028c122a/AUTHORS) from upstream. This may be legally required to meet the LGPL’s obligations, depending on your interpretation of what counts as a “copyright notice”, but it is in any case certainly the right thing to do. You can of course add yourself and @cclauss to the list.

Reply:
> The original LGPL code and the newer MIT code are intertwined.

There is no "newer" MIT code. This is why the GLP and related copy-left licenses are sometimes referred to as "viral licenses", because they infect everything they touch. If you contribute things intertwined in an LGPL code bases your contributions are *automatically* also LGPL licensed whether you want them to be or not. As the author you can additionally contribution those ports as MIT (provided they can be distributed and used separately), but you cannot say that they cannot be used as LGPL.